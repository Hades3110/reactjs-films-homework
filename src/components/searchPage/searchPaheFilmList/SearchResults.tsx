import { useState, useEffect } from 'react'
import FilmItem from '../../main/filmsList/filmItem/FilmItem'
import {
    FilmInterface,
    getGenres,
    GenreInterface,
    getSearchResult
} from '../../../services/filmService'
import spinner from '/public/assets/spinner.gif'
import { useLocation } from 'react-router'
import styles from './searchResults.module.scss'

const Searchresults = () => {

    const [filmArr, setFilmArr] = useState<FilmInterface[]>([])
    const [genres, setGenres] = useState<GenreInterface[]>([])
    const [filmNotFount, setFilmNotFount] = useState<boolean>(false)
    const [isLoaded, setIsLoading] = useState<boolean>(false)

    const { search } = useLocation()
    const searchParams = new URLSearchParams(search)
    const film = searchParams.get('film')

    useEffect(() => {
        getSearchResult(film ?? '')
            .then(res => {
                const result = res.results.filter((film, index, self) =>
                    self.findIndex((t) => (
                        t.id === film.id
                    ))
                )
                if (result.length > 0) {
                    setFilmNotFount(false)
                    setFilmArr(result.filter(film => film.poster_path && film.title))
                    setIsLoading(true)
                } else {
                    setIsLoading(true)
                    setFilmNotFount(true)
                }
            })
    }, [film])

    useEffect(() => {
        getGenres()
            .then(res => setGenres(res.genres))
    }, [])

    return (
        <>
            {isLoaded ? <img src={spinner} className={styles.spinner} /> :
                (filmNotFount ? <div className={styles.searchErrorMassage}>There are no movies that matched your query.</div> :
                    <div className={styles.filmList}>
                        <div className='filmListRow'>
                            {
                                filmArr.map((el: FilmInterface) => {
                                    return (
                                        <FilmItem
                                            key={el.id}
                                            id={el.id}
                                            title={el.title === '' ? 'NAME NOT FOUND' : el.title}
                                            image={el.poster_path}
                                            rate={el.vote_average}
                                            overview={el.overview === '' ? 'No Description' : el.overview}
                                            filmGenres={el.genre_ids}
                                            genres={genres}
                                        />
                                    )
                                })
                            }
                        </div>
                    </div>
                )
            }
        </>
    )
}

export default Searchresults
