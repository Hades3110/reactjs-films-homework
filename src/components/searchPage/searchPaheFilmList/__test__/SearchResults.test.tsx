import { create } from "react-test-renderer";
import SearchResults from '../SearchResults';
import { BrowserRouter as Router } from 'react-router-dom'

describe('Footer Component', () => {

  it('Footer snapshot test', () => {
    const component = create(<Router><SearchResults /></Router>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});