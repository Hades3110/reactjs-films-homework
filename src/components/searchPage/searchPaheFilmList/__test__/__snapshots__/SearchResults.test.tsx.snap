// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component Footer snapshot test 1`] = `
<img
  className="spinner"
  src="test-file-stub"
/>
`;
